framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -821,19 +894,21 @@ public Boolean visitTypevar_Typevar(
         // as the bounds are the same.
         return isPrimarySubtype(subtype, supertype);
 
-      } else if (!subtypeHasAnno && !supertypeHasAnno && areEqualInHierarchy(subtype, supertype)) {
+      } else if (!subtypeHasAnno && !supertypeHasAnno) {
         // two unannotated uses of the same type parameter are of the same type
-        return true;
-
-      } else if (subtype.getUpperBound().getKind() == TypeKind.INTERSECTION) {
-        // This case happens when a type has an intersection bound.  e.g.,
-        // T extends A & B
-        //
-        // And one use of the type has an annotation and the other does not. e.g.,
-        // @X T xt = ...;  T t = ..;
-        // xt = t;
-        //
-        return visit(subtype.getUpperBound(), supertype.getLowerBound(), null);
+        return areEqualInHierarchy(subtype, supertype);
```

Comment:
Above, [line 815/888](https://github.com/typetools/checker-framework/pull/3922/files#diff-c1fb721f944f7a0dc56d06e7c62d6e738492ce93a4c9d6bfb23f62491eb5d03fR888) says "// subtype and supertype are uses of the same type parameter".  If that is true, why cannot this line be just `return true;`?  Maybe a comment would help to clarify.